test/fennec/udp/data_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule Fennec.UDP.DataTest do
+  use ExUnit.Case, async: false
+  use Helper.Macros
+
+  alias Helper.UDP
+  alias Jerboa.Params
+  alias Jerboa.Format
+  alias Jerboa.Format.Body.Attribute.{Data, XORPeerAddress, XORRelayedAddress}
+
+  import Mock
+
+  @peer_addr {127, 0, 0, 1}
+
+  setup ctx do
+    {:ok, [udp: UDP.setup_connection(ctx, :ipv4)]}
+  end
+
+  describe "incoming peer datagram" do
+
+    setup ctx do
+      params = UDP.allocate(ctx.udp)
+      {:ok, [relay_sock: Params.get_attr(params, XORRelayedAddress)]}
+    end
+
+    test "gets discarded when there's no permission for peer", ctx do
+      ## given a relay address for a peer with no permission
+      %XORRelayedAddress{address: relay_addr, port: relay_port} = ctx.relay_sock
+      {:ok, peer} = :gen_udp.open(0, [{:active, :false}, :binary])
+      {:ok, peer_port} = :inet.port(peer)
+      self_ = self()
+      with_mock Fennec.UDP.Worker, [:passthrough], [
+        handle_peer_data: fn (:no_permission, ip, port, data, state) ->
+          ## we can't use `assert called ...` as we want to ignore `state`
```

Comment:
`assert called ...` only calls `:meck.called(...)` - it's not a macro, but a function call, therefore it requires a proper value (not a wildcard) for each of the passed in arguments.